Unix/http/httpclient.c

```diff
@@ -729,20 +639,33 @@ static Http_CallbackResult _ReadHeader(
         memcpy(handler->recvPage + 1, data, handler->receivedSize);
     handler->recvingState = RECV_STATE_CONTENT;
 
-    /* Invoke user's callback with header information */
+    /* Check the authentication. If we need to recycle, send a response to the response. */
+
+    if (!handler->isAuthorized) 
```

Comment:
what if the client wants to reauthenticate the connection? This decision needs to be 
- if we have a password, or
- we have not authorized the connection
then do authorization.